include/mata/nfa/nfa.hh

```diff
@@ -742,7 +747,7 @@ bool is_prfx_in_lang(const Nfa& aut, const Run& word);
  // TODO: rename to something, but no idea to what.
  // Maybe we need some terminology - Symbols and Words are made of numbers.
  // What are the symbol names and their sequences?
-Run encode_word(const StringToSymbolMap& symbol_map, const std::vector<std::string>& input);
+Run encode_word(const Alphabet* alphabet, const std::vector<std::string>& input);
```

Comment:
good that alphabet can be used here

Reply:
Btw the above comment about terminology is still valied I think.

Can you add some comment on what input is? vector of strings does not look like a word

Reply:
A vector of strings is what we use for words in Mata. Should that be a type alias? I will add a comment here.

Reply:
Why is a word vector of strings? It should be a vector of symbols, no?

Reply:
It is a vector of strings, where each string corresponds to a single symbol in a user-space. Internally, we would work with words as a vector of symbol values (`Symbol` type) which is a library-space representation of a word. This function is an interface function, expecting to accept words from the user-space and working with them internally. This does not make sense for alphabets such as `EnumAlphabet` or `ImplicitAlphabet`. It does make sense for `StringAlphabet`. However, it is a necessary evil of inheritance (we need a common interface). We can try to create a more sensible representation. This solution is far from perfect.

Reply:
I don't get it ,it sounds weird, lets discuss it online.

Reply:
We decided to name a vector of symbol a word, and a vector of symbol names a word name. This will be implemented in a future PR.

Reply:
Types `Word` and `WordName` were added.